src/learn/DOTNET/index.md

```diff
@@ -0,0 +1,155 @@
+
+```
+
+
+
+## Personalizing your project
+
+Open the ne directory and navigate to the program file. This file is the entrypoint for execution and where servers and middleware can be configured. When you create a new project microsoft helpfully includes some code to tell you the weather in this project. We can delete this. After removing the weather stuff the file should look like
+
+```cs
+
+var builder = WebApplication.CreateBuilder(args);
+
+// Add services to the container.
+// Learn more about configuring Swagger/OpenAPI at https://aka.ms/aspnetcore/swashbuckle
+builder.Services.AddEndpointsApiExplorer();
+builder.Services.AddSwaggerGen();
+
+var app = builder.Build();
+
+// Configure the HTTP request pipeline.
+if (app.Environment.IsDevelopment())
+{
+    app.UseSwagger();
+    app.UseSwaggerUI();
+}
+
+app.UseHttpsRedirection();
+
+
+
+app.Run();
+
+
+```
+before we leave program to make an api controler we want to add 3 more lines so the program files.
+1. builder.Services.AddControllers();
+ so the file knows we are making a controller 
+2. app.UseAuthorization();
+which is middleware that is required to run on a route before it acesses the endpoint
+3.app.MapControllers();
+which attatches endpoints to our controllers.
+
+the final file should look like this
+```cs
+var builder = WebApplication.CreateBuilder(args);
+
+// Add services to the container.
+// Learn more about configuring Swagger/OpenAPI at https://aka.ms/aspnetcore/swashbuckle
+builder.Services.AddControllers();
+builder.Services.AddEndpointsApiExplorer();
+builder.Services.AddSwaggerGen();
+
+var app = builder.Build();
+
+// Configure the HTTP request pipeline.
+if (app.Environment.IsDevelopment())
+{
+    app.UseSwagger();
+    app.UseSwaggerUI();
+}
+
+app.UseHttpsRedirection();
+
+app.UseAuthorization();
+
+app.MapControllers();
+
+
+app.Run();
+
+
+```
+
+
+## Creating a controller
+
+
+inside your MyWebApi directory make a new directory called controllers. Inside this directory make a new file called MyController.cs
+
+within the file add this code
+```cs
+// imports useful methods for setting up a controller
+using Microsoft.AspNetCore.Mvc;
+// sets up an api controller 
+namespace MyApi.Controllers
+{
+    [ApiController]
+    [Route("[controller]")]
+    public class HelloController : ControllerBase
+    {
+// creates our get route
+        [HttpGet]
+        public string Get()
+        {
+          //decides what is returned by the route
+            return "Hello, World!";
+        }
+    }
+}
+
+
+```
+you now have a very simple api built in dotnet
+## Using Swagger to test controllers
+
+make sure your terminal is inside the api directory then use
+
+```shell
+dotnet run
+```
+You should get a link to visit but there is nothing there. The trick is to add /swagger onto the end of your url. If everthing has worked properly you should be taken to a web page where you can click around and test any routes from your controller give it a go.
```

Comment:
```suggestion
You should get a link to visit but there is nothing there. The trick is to add `/swagger` onto the end of your url. If everything has worked properly you should be taken to a web page where you can click around and test any routes from your controller give it a go.
```

add a screenshot here of what swagger needs to look like for them - go on, do it !